### PR TITLE
NET-6050, parse simple Grok patterns w/o syslog processors being invo…

### PR DESCRIPTION
**Pull request by ab0ndar**
NET-6050, parse simple Grok patterns w/o syslog processors being involved

### CODE REVIEW

File: nsgcli/grok.py [hunk 1]
```diff
@@ -92,10 +94,49 @@ def do_parse(self, _):
...
+                                       't:p:',
+                                       ['text=', 'pattern='])
+        except getopt.GetoptError as ex:
+            print('UNKNOWN: Invalid Argument:' + str(ex))
+            raise InvalidArgsException
+
+        txt = None
+        pattern = None
+        for opt, arg in opts:
+            if opt in ['-t', '--text']:
```
Comment (vkurland) [inline nsgcli/grok.py]: this looks strange. Why reach for `sys.argv` again if you should be getting parameters as a second parameter in the call to `do_log()` and `do_text()` ?

also I do not know how do I feel about using getopt inside of the command (i.e. `text --pattern ... --text`) . This contradicts general design of the CLI in nsgcli that is based around Python module Cmd. Arguments of other commands are positional and are not preceded with "--word" .

**Review by vkurland — CHANGES_REQUESTED** [review]

File: nsgcli/nsggrok_main.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        this function finds and returns the value of the key 'error'. Otherwise it returns
+        the whole response as a string
+        """
+        if isinstance(response, dict):
+            return response.get('error', str(response))
+        else:
+            return str(response)
+
+    def call_grok_api(self, parser, txt, pattern):
+        request = 'v2/grok/net/{0}/parser/{1}'.format(self.netid, parser)
```
Comment (vkurland) [inline nsgcli/nsggrok_main.py:117]: I believe the value of the parameter `pattern` is always `self.pattern`, if so, then function `call_grok_api()` does not really need this parameter

File: nsgcli/nsggrok_main.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+    def do_text(self, cmd_args):
+        """
+        Parse text line with grok patterns.
+        Require 'pattern' parameter.
+
+        --pattern="Hello world of %{WORD:world_name}" text <text>
+
+        Parameters:
+            text - Text to be parsed, read stdin if not specified
+
```
Comment (vkurland) [inline nsgcli/nsggrok_main.py:90]: it looks like you support multi-line input where each line is parsed separately (but I am 100% sure I read this right). If so, please reflect this in the docstring and script "usage" text

File: nsgcli/nsggrok_main.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+        log <message>
+
+        Parameters:
+            message - Syslog Message to be parsed, read stdin if not specified
+
+        Examples:
+            log "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host '10.1.1.1'"
+            log <<< "<13>May 18 11:22:43 carrier sshd: SSHD_LOGIN_FAILED: Login failed for user 'root' from host 10.1.1.1"
+        """
```
Comment (vkurland) [inline nsgcli/nsggrok_main.py:73]: for consistency, make these examples follow the same format as those in `do_text()`, that is, show how to provide the pattern. Also, add these examples to the script's "usage" text

**Review by vkurland — CHANGES_REQUESTED** [review]

File: nsgcli/nsggrok_main.py (hunk 2 above)
Comment (ab0ndar) [reply nsgcli/nsggrok_main.py:117]: right, simplified.

File: nsgcli/nsggrok_main.py (hunk 3 above)
Comment (ab0ndar) [reply nsgcli/nsggrok_main.py:90]: added example with stdin/multiline

File: nsgcli/nsggrok_main.py (hunk 4 above)
Comment (ab0ndar) [reply nsgcli/nsggrok_main.py:73]: for the 'log' command 'pattern' parameter is optional, I've added one example with 'pattern'

**Review by vkurland — APPROVED** [review]
LGTM

**Pull request merged**